Spring-server/src/main/java/ikpi63holding/postclient/ImapConnector.java

```diff
@@ -0,0 +1,140 @@
+package ikpi63holding.postclient;
+
+import java.io.IOException;
+import java.security.KeyManagementException;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import javax.mail.Flags;
+import javax.mail.Folder;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Store;
+import javax.net.ssl.KeyManager;
+import javax.net.ssl.SSLContext;
+import javax.net.ssl.TrustManager;
+import javax.net.ssl.X509TrustManager;
+
+public class ImapConnector {
+    private static String IMAPS_PORT = "993";
+
+    static {
+        SSLContext ctx = null;
+        try {
+            ctx = SSLContext.getInstance("TLS");
+        } catch (NoSuchAlgorithmException e) {
+            System.out.println(e.toString());
+        }
+        try {
+            ctx.init(new KeyManager[0], new TrustManager[] {new DefaultTrustManager()},
+                    new SecureRandom());
+        } catch (KeyManagementException e) {
+            System.out.println(e.toString());
+        }
+        SSLContext.setDefault(ctx);
+    }
+
+    private Session session;
+    private Store store;
+
+    ImapConnector(String login, String password, String host) throws Exception {
+
+        this.session = getImapSession(login, password);
+        this.store = this.session.getStore();
+        store.connect(host, Integer.parseInt(IMAPS_PORT), login, password);
+    }
+
+    public Session getImapSession(String login, String password) throws Exception {
+
+        if (login == null) {
+            throw new Exception("No such email");
+        }
+        Properties properties = new Properties();
+        properties.put("mail.debug", "false");
+        properties.put("mail.store.protocol", "imaps");
+        properties.put("mail.imap.ssl.enable", "true");
+        properties.put("mail.imap.port", IMAPS_PORT);
+
+        Session sessionImap =
+                Session.getDefaultInstance(properties, new javax.mail.Authenticator() {
+                    protected PasswordAuthentication getPasswordAuthentication() {
+                        return new PasswordAuthentication(login, password);
+                    }
+                });
+        return sessionImap;
+    }
+
+    public List<String> getFolders() throws MessagingException {
+        return getAllFolders(this.store.getDefaultFolder());
+    }
+
```

Comment:
What difference between getFolders and getAllFolders? I think getFolders is unnecessary.

Reply:
Get all folders is an internal method operaiting with JavaMail Folder object,
since I don't want to move any JavaMail object outside our mail and connector classes, I exposed this functionality through  parameterless method, inside witch I use store and get it's root folder to pass next.

Maybe we should think about correct naming, but I insist that both methods are serving a purpose.